src/coredump/coredump.c

```diff
@@ -1656,28 +1660,35 @@ static int forward_coredump_to_container(Context *context) {
         return 0;
 }
 
-static int gather_pid_mount_tree_fd(const Context *context) {
+static int gather_pid_mount_tree_fd(const Context *context, int *ret_fd) {
         _cleanup_close_ int mntns_fd = -EBADF, root_fd = -EBADF;
         _cleanup_close_pair_ int pair[2] = EBADF_PAIR;
         int fd = -EBADF, r;
-        pid_t child;
 
         assert(context);
+        assert(ret_fd);
 
         /* Don't bother preparing environment if we can't pass it to libdwfl. */
 #if !HAVE_DWFL_SET_SYSROOT
-        return -EOPNOTSUPP;
+        r = 0;
+        goto finish;
 #endif
```

Comment:
the rest of the function should probably be guarded by an `#else` branch

Reply:
unaddressed.